datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -34,8 +34,65 @@ use datafusion_expr::ColumnarValue;
 
 use itertools::izip;
 
-/// Expression that can be evaluated against a RecordBatch
+/// `PhysicalExpr` evaluate DataFusion expressions such as `A + 1`, or `CAST(c1
+/// AS int)`.
+///
+/// `PhysicalExpr` are the physical counterpart to [`Expr`] used in logical
+/// planning, and can be evaluated directly on a [`RecordBatch`]. They are
+/// normally created from `Expr` by a [`PhysicalPlanner`] and can be created
+/// directly using [`create_physical_expr`].
+///
 /// A Physical expression knows its type, nullability and how to evaluate itself.
+///
+/// [`PhysicalPlanner`]: https://docs.rs/datafusion/latest/datafusion/physical_planner/trait.PhysicalPlanner.html
+/// [`create_physical_expr`]: crate::create_physical_expr
+/// [`Expr`]: datafusion_expr::Expr
+///
+/// # Example: Create `PhysicalExpr` from `Expr`
+/// ```
+/// # use arrow_schema::{DataType, Field, Schema};
```

Comment:
This looks pretty terrible in source form, but it renders reasonably I think

<img width="1059" alt="Screenshot 2024-02-09 at 7 57 06 AM" src="https://github.com/apache/arrow-datafusion/assets/490673/d172a057-62c2-46f9-bec2-4b4b49228981">